packages/react-native/ReactAndroid/src/main/java/com/facebook/react/views/scroll/ReactHorizontalScrollView.java

```diff
@@ -1413,13 +1411,29 @@ private void adjustPositionForContentChangeRTL(int left, int right, int oldLeft,
         float flingVelocityX = mScroller.getCurrVelocity() * direction;
 
         mScroller.fling(
-            scrollX, getScrollY(), (int) flingVelocityX, 0, 0, newWidth - getWidth(), 0, 0);
+          scrollX, getScrollY(), (int) flingVelocityX, 0, 0, maxX, 0, 0);
       } else {
         scrollTo(scrollX + (mScroller.getCurrX() - scrollerXBeforeTick), getScrollY());
       }
     }
   }
 
+  private void adjustPositionForContentChangeRTL(int left, int right, int oldLeft, int oldRight) {
+    // If we have any pending custon flings (e.g. from aninmated `scrollTo`, or flinging to a snap
+    // point), finish them, commiting the final `scrollX`.
+    // TODO: Can we be more graceful (like OverScroller flings)?
+    if (getFlingAnimator().isRunning()) {
+      getFlingAnimator().end();
+    }
```

Comment:
Is this piece still relevant? I.e. if adjustment happens during programmatically triggered fling?

Reply:
I think it would be pretty rare cases, but I added the extra logic just in case.

There I used cancel instead of end since I do not want to commit the last scroll position of the animation. I think it is better to cancel the scroll still than to have content jumps because of the animation being wrong.